flake.nix

```
@@ -0,0 +1,27 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+  };
+
+  outputs = {nixpkgs, ...}: let
+    system = "x86_64-linux";
+    pkgs = nixpkgs.legacyPackages.${system};
+  in {
+    devShells.${system}.default = pkgs.mkShell {
+      nativeBuildInputs = with pkgs; [
+        cargo-binstall
+        dioxus-cli
+        wasm-bindgen-cli_0_2_100
+      ];
+
+      # https://docs.rs/getrandom/0.3.3/getrandom/#webassembly-support
+      RUSTFLAGS = "--cfg getrandom_backend=\"wasm_js\"";
+
+      # gcc is default
```

suggestion (blocking): I did some more research but couldn't really figure out why gcc doesn't work and where I got the information from that clang does work. but maybe we can still be a bit more detailed here:

```suggestion
      # with gcc (default), the WASM build fails silently
      # at runtime there is an error message similar to https://github.com/DioxusLabs/dioxus/discussions/3807
      # maybe this can be resolved with dioxus 0.7.0 due to https://github.com/DioxusLabs/dioxus/issues/4023
```

---

after upgrading dioxus as a dependencies **and** dioxus-cli on the system to 0.7.1, I now get a different error when using gcc.

`ethersync-web.js` contains the line `import * as __wbg_star1 from 'env';` which refers to a non-existing module `env`.

at runtime this causes

> Uncaught TypeError: Error resolving module specifier “env”. Relative module specifiers must start with “./”, “../” or “/”.

which results in search hits with very different causes and solutions.

switching to clang again resolves the problem magically… :shrug:

---

Ah, let's stick to clang then, and re-investigate at some other point :)